### ISSUE TITLE
Mac OS X Install failure

### ISSUE THREAD

**Issue by crose1000** [eXist-db/exist#1847]
Thank you for reporting your issue!

To be able to better understand you problem or suggestion, please add as much information as possible to this ticket. When there is no or just little information, it will be too difficult to work on it (and hence we will not be able to solve it)

Please fill in the following sections:

### What is the problem
> Describe exactly what you see (e.g. an output of an XQuery) 
Run installer for 4.1.0 on OS X High Sierra 10.13.3, either via jar or dmg.
Launch app, either as service or via double-clicking.
Go to localhost:8080 in browser, re-directs to dashboard ==> 
javax.servlet.ServletException: javax.servlet.ServletException: java.lang.SecurityException: class "javax.annotation.Nullable"'s signer information does not match signer information of other classes in the same package

### What did you expect
> Describe what you expected to happen. Add for example a reference to a [specification](https://www.w3.org/TR/xquery-3/).
Expected to see the Dashboard

### Describe how to reproduce or add a test
> Describe how we can can reproduce the problem.
Run installer and launch exist.

> The *best* way is to provide a 'self containing' test, a test that can be run without dependancies. The [XQSuite - Annotation-based Test Framework for XQuery](http://exist-db.org/exist/apps/doc/xqsuite.xml) makes it very easy for you to create tests. These tests can be executed from the [eXide editor](http://exist-db.org/exist/apps/eXide/index.html) (XQuery - Run as Test)
Your installer contains all required assets to reproduce bug.

### Context information
Please always add the following information
- eXist-db version + Git Revision hash e.g. eXist-db 3.0 /  acd0c14 
4.1.0
- Java version (e.g. Java8u121)
java version "9.0.1"
Java(TM) SE Runtime Environment (build 9.0.1+11)
Java HotSpot(TM) 64-Bit Server VM (build 9.0.1+11, mixed mode)
- Operating system (Windows 7, Linux, MacOs)
MacOs 10.13.3
- 32 or 64 bit
64 bit
- Any custom changes in e.g. conf.xml
None

**Comment by crose1000** [comment]
Here's the same bug, entirely different project. It may provide some useful background info.
https://github.com/Microsoft/ApplicationInsights-Java/issues/362

**Comment by dizzzz** [comment]
thank you for your analysis ; 

We do sign jars for the webstart function, but the betterform jar is not part of this. (only the lib/core)

I am wondering why did not receive other reports, as many of the devs run and test and macOS and myself I installed a production server with 3.1 using the jar file, and the server is fully functional.

note that java9.0.1 was quickly replaced with 9.0.4 ; It is recommended for now to run eXist-db with java8 (a lot of discussions recently). Please could you try that? 

@duncdrum

**Comment by crose1000** [comment]
Set java to java 8 in startup.sh unconditionally, same behavior.
#if [ ! -d "$JAVA_HOME" ]; then
#	JAVA_HOME="/Library/Java/JavaVirtualMachines/jdk-9.0.1.jdk/Contents/Home"
#fi
JAVA_HOME="/Library/Java/JavaVirtualMachines/jdk1.8.0_151.jdk/Contents/Home"
Plus, java control panel indicates I do have 9.0.4 installed. Profile sets JAVA_HOME
export JAVA_HOME=$(/usr/libexec/java_home)
So I do not understand this behavior and will need to explore.

**Comment by duncdrum** [comment]
@dizzzz yes this is consistent with trying to start exist when JAVA_HOME is pointing to java9 (on win or mac as far as i can tell).
@crose1000 can you confirm when you type into terminal: 
```bash
echo $JAVA_HOME
```
it points to java8 (you might need to relog your OS profile). For me these errors disappear when the installer is run with JAVA_HOME pointing to 8.

**Comment by crose1000** [comment]
I changed in the exist startup.sh script, not in my environment. In the terminal:
MITP6:~ crose$ java -version
java version "9.0.1" ...
I will re-run the installer with java 8 and report results.

**Comment by shanemcgarry** [comment]
I have the same issue on Arch Linux distro. It seems to be specific to version 4.1.0. I was able to install and run version 4.0.0 without issue.

**Comment by crose1000** [comment]
I am now able to install and run successfully, using the dmg, running Java 8. I am checking the jar installer for Java 8 now.

**Comment by duncdrum** [comment]
@shanemcgarry java9 & java10 won't work, because of changes to the java versioning scheme, for now exist requires java8.

@crose1000 sounds promising

**Comment by duncdrum** [comment]
Since everybody here seems to be up and running again, I propose to close this. 
The state of Java9 support can be tracked in #1792

**Comment by crose1000** [comment]
That's fine with me. Since there were some others weighing in on other platforms I thought I would wait until everyone had chimed in.
________________________________________
